skrobot/planner/utils.py

```diff
@@ -0,0 +1,192 @@
+
+    if with_base:
+        av_joint, av_base = av[:-3], av[-3:]
+        x, y, theta = av_base
+        co = Coordinates(pos=[x, y, 0.0], rot=rpy_matrix(theta, 0.0, 0.0))
+        robot_model.newcoords(co)
+    else:
+        av_joint = av
+
+    for joint, angle in zip(joint_list, av_joint):
+        joint.joint_angle(angle)
+
+
+def get_robot_config(robot_model, joint_list, with_base=False):
+    """A utility function for getting robot state
+
+    Parameters
+    ----------
+    robot_model : skrobot.model.CascadedLink
+        robot model
+    joint_list : list[Joint]
+        joint list of which you want to know the angles
+    with_base : bool
+        If set to `True`, base position is also computed.
+
+    Returns
+    -------
+    av_whole (or av_joint) : numpy.array(n_dof,)
+        angle vector. If `with_base=False`, `n_dof` is the number of
+        joints `n_joint`, but if `with_base=True`, `n_dof = n_joint + 3`.
+    """
+    av_joint = np.array([j.joint_angle() for j in joint_list])
+    if with_base:
+        x, y, _ = robot_model.translation
+        rpy = rpy_angle(robot_model.rotation)[0]
+        theta = rpy[0]
+        av_whole = np.hstack((av_joint, [x, y, theta]))
+        return av_whole
+    else:
+        return av_joint
+
+
+def forward_kinematics_multi(robot_model,
+                             joint_list,
+                             av,
+                             move_target_list,
+                             with_rot,
+                             with_base,
+                             with_jacobian=True):
+    """Compute fk for multiple feature points
+
+    Parameters
+    ----------
+    robot_model : skrobot.model.CascadedLink
+        robot model.
+    joint_list : list[skrobot.model.Joint]
+        joint to be controlled
+    av : numpy.ndarray(n_dof,)
+        angle vector.
+    move_target_list : list[skrobot.coordinates.CascadedCoords]
+        the list has `n_feature` elements. Each element is the
+        coordinate of the features points.
+    with_rot : bool
+        If set to `True`, 7(3 + 4) dim pose-fk is also computed.
+        Otherwise, 3 dim point-fk is computed.
+    with_base : bool
+        If `with_base=False`, `n_dof` is the number of joints `n_joint`,
+        but if `with_base=True`, `n_dof = n_joint + 3`.
+
+    Returns
+    -------
+    pose_arr : numpy.ndarray(n_feature, dim_pose)
+        array of pose of each feature points.
+        `dim_pose=7' if `with_rot=True`. Otherwise, `dim_pose=3`.
+    jac_arr : numpy.ndarray(n_feature, dim_pose, n_dof)
+        array of jacobian of each feature points.
+
+    """
+
+    set_robot_config(robot_model, joint_list, av, with_base)
+    n_feature = len(move_target_list)
+    n_dof = len(av)
+    dim_pose = 3 + (4 if with_rot else 0)
+
+    pose_arr = np.zeros((n_feature, dim_pose))
+    jac_arr = np.zeros((n_feature, dim_pose, n_dof))
+
+    for i in range(n_feature):
+        pose, jac = _forward_kinematics(
```

Comment:
This occurs an error when `with_jacobian` is `False`.

Reply:
thanks!I will fix it and add some tests. , wait for moment.

Reply:
fixed and tested in https://github.com/iory/scikit-robot/pull/173/commits/c756142176568a3ce3d8c27a023cf24be348c089